menpo/shape/mesh/base.py

```diff
@@ -236,6 +309,257 @@ def n_tris(self):
         """
         return len(self.trilist)
 
+    def half_edges(self):
```

Comment:
This should definitely have a test - and would be good to move into new PR

I feel like this would be extremely slow - is this the kind of code that would benefit from Cython or even from some JIT code or something?

Reply:
Yup,  you are totally right about cython as the original code  is written in cython (https://github.com/joel-simon/cymesh).  We can discuss in an new PR if we like to have this feature in menpo and how we will have it.